core/src/progress_map.rs

```diff
@@ -351,6 +351,14 @@ pub struct PropagatedStats {
 }
 
 impl PropagatedStats {
+    pub fn get_latest_leader_slot(&self, slot_of_self: Slot) -> Option<Slot> {
```

Comment:
This seems confusing to add another `get_latest_leader_slot` which may return the slot provided. Could this be handled in the ProgressMap function?

Reply:
done